### ISSUE TITLE
"failed integrity check" when attempting to list videos

### ISSUE THREAD

**Issue by KoenDG** [ihabunek/twitch-dl#124]
Example: 
```
twitch-dl.2.1.2.pyz videos macaw54
GraphQL query failed
* failed integrity check
* ```

Download functionality seems fine.

**Comment by Kloenk** [comment]
see the same, debug flag does not show much more
```
DEBUG:httpx._client:HTTP Request: POST https://gql.twitch.tv/gql "HTTP/1.1 200 OK"
GraphQL query failed
* failed integrity check

**Comment by arrowgent** [comment]
aye, updated 2.1.2 python3.7 to verify issue exists

**Comment by vvibee** [comment]
Same problem, getting vod info and downloading vod by given vod id is working fine, but can not get channel vods list by streamer's login id.
twitch gql response: "failed integrity check".

**Comment by NotQuiteApex** [comment]
Running into a similar issue with my own program (not twitch-dl).

Twitch seems to have rolled out a new integrity system regarding requests to GQL. Certain query data is restricted without passing an integrity check. Sending a POST request to `https://gql.twitch.tv/integrity` with the Client-Id header gets you the necessary token, an expiration date, and a request ID. I have yet to figure out what to do with this yet, but it appears necessary for new GQL requests.

**Comment by NotQuiteApex** [comment]
Found the culprit. It has nothing to do with Twitch's new integrity endpoint or new headers. It has to do with empty strings originally evaluating to null when used in the context of pagination arguments. The GQL query related to getting channel videos uses an empty string to be equivalent to null when its the first query before paging through more, see https://github.com/ihabunek/twitch-dl/blob/5a43a4388c51c345c59c046a2b04a3b1e8ef0eec/twitchdl/twitch.py#LL233C29-L233C29. There is probably a simple fix for this.

**Comment by vvibee** [comment]
@NotQuiteApex
@KoenDG 
@Kloenk 
@arrowgent 
For now, the simple fix for this problem is change 
client-id: "kimne78kx3ncx6brgo4mv6wki5h1ko" -> "kd1unb4b3q4t58fwlpcbzcbnm76a8fp" (Both are public IDs)
The old client-id may be restricted, take it from here: https://github.com/lay295/TwitchDownloader/commit/be2c9f1ac5b51f88ee1f149e70f05f22a1d6f23a

**Comment by ihabunek** [comment]
Thanks to everybody for debugging this and @mwhite34 for the pull request! This fix is now available in 2.1.3.

I dread to think what hoops twitch will make us jump next. :)

**Comment by arrowgent** [comment]
id gladly use my own bot's oauth token... or any specific application that im using

but thanks for recognizing that the old twitch-dl token was "expired"
